### DISCUSSION TITLE
Day 2: I cannot access my table. Shows blank...?

### DISCUSSION THREAD

**Discussion by elizabethhannan** [AntlerVC/firetable#1408] [Support / Q&A] [answered]
Hi, 
I am following this setup tutorial Day 1. (https://www.youtube.com/watch?v=CQuhFFBxGY0)
Uploaded a CSV yesterday.  Today in Rowy I get a popup error message that I don't have access to my own table. No table valuables are displaying. (No other users.)
How do I fix this? I have checked permissions and settings and it all shows that I should have access. 

2nd issue:
Now with derivative function (also in Rowy support video) **"Build Fails"** error code log below: (Identical to video tutorial).

Thanks,
-E

Build failed
0
17:51:52:52  Build started. collectionType: collection, tablePath: keystrokes, pathname: table/keystrokes, functionName: keystrokes

1
17:51:53:53  Duplicating functions template to build/functionBuilder/builds/1694393512994

2
17:51:53:53  

3
17:51:53:53  Executing: cd build/functionBuilder/builds/1694393512994;pwd;ls -al

4
17:51:53:53  stdout: {"error":null,"stdout":"/home/node/app/build/functionBuilder/builds/1694393512994
total 8
drwxrwxrwx    2 root     root           140 Sep 11 00:51 .
drwxrwxr-x    2 root     root            60 Sep 11 00:51 ..
-rw-r--r--    1 root     root            43 Jul  2 06:45 firebase.json
-rw-r--r--    1 root     root           596 Jul  2 06:45 package.json
drwxr-xr-x    2 root     root           320 Jul  2 14:28 src
-rw-r--r--    1 root     root           409 Jul  2 06:45 tsconfig.json
-rw-r--r--    1 root     root          5131 Jul  2 06:45 tslint.json
","stderr":""}

5
17:51:53:53  1

6
17:51:53:53  Executing: cd build/functionBuilder/builds/1694393512994;less package.json

7
17:51:53:53  stdout: {"error":null,"stdout":"{
  "name": "rowy-functions",
  "version": "1.0.0",
  "scripts": {
    "deploy": "echo 'n
' | firebase deploy --interactive"
  },
  "engines": {
    "node": "18"
  },
  "main": "src/index.js",
  "dependencies": {
    "@google-cloud/logging": "^10.3.1",
    "@google-cloud/secret-manager": "^4.1.4",
    "@google-cloud/storage": "6.7.0",
    "file-type": "^16.5.3",
    "firebase-admin": "^11.3.0",
    "firebase-functions": "^4.1.0",
    "lodash": "^4.17.21",
    "node-fetch": "^2.6.7",
    "uuid": "^8.3.2",
    "google-auth-library": "8.8.0"
  },
  "devDependencies": {},
  "private": true
}
","stderr":""}

8
17:51:53:53  2

9
17:51:53:53  Executing: cd build/functionBuilder/builds/1694393512994/src;pwd;ls -al

10
17:51:53:53  stdout: {"error":null,"stdout":"/home/node/app/build/functionBuilder/builds/1694393512994/src
total 12
drwxr-xr-x    2 root     root           320 Jul  2 14:28 .
drwxrwxrwx    2 root     root           140 Sep 11 00:51 ..
drwxr-xr-x    2 root     root            80 Jul  2 14:28 derivatives
drwxr-xr-x    2 root     root            80 Jul  2 14:28 extensions
-rw-r--r--    1 root     root           704 Jul  2 14:28 firebaseConfig.js
-rw-r--r--    1 root     root           495 Jul  2 14:28 firebaseConfig.js.map
-rw-r--r--    1 root     root           215 Jul  2 14:28 functionConfig.js
-rw-r--r--    1 root     root           221 Jul  2 14:28 functionConfig.js.map
-rw-r--r--    1 root     root          3874 Jul  2 14:28 index.js
-rw-r--r--    1 root     root          2190 Jul  2 14:28 index.js.map
drwxr-xr-x    2 root     root            80 Jul  2 14:28 initialize
drwxr-xr-x    2 root     root            80 Jul  2 14:28 logging
drwxr-xr-x    2 root     root           160 Jul  2 14:28 propagates
-rw-r--r--    1 root     root          1446 Jul  2 14:28 rowy.js
-rw-r--r--    1 root     root          1102 Jul  2 14:28 rowy.js.map
drwxr-xr-x    2 root     root           200 Jul  2 14:28 utils
","stderr":""}

11
17:51:53:53  3

12
17:51:53:53  Executing: cd build/functionBuilder/builds/1694393512994;yarn install --mutex network

13
17:52:09:09  stdout: {"error":null,"stdout":"info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
[5/5] Building fresh packages...
success Saved lockfile.
","stderr":""}

14
17:52:09:09  Base dependencies installed successfully

15
17:52:09:09  Generating schema...

16
17:52:09:09  derivativeColumns: [{"fieldName":"test","type":"DERIVATIVE","key":"test","config":{"derivativeFn":"// Import any NPM package needed
// import _ from "lodash";

const derivative: Derivative = async ({ row, ref, db, storage, auth, logging }) => {
  logging.log("derivative started");

};

//write your code below here only

const url = row.image;
const image = await rowy.storage.upload.url(url);
ref.update({"image" : logo.image.downloadURL});

return [image];

//write your code above here only

export default derivative;","listenerFields":["logoImage"],"renderFieldType":"IMAGE"},"index":5,"name":"imagefile"}]

17
17:52:09:09  defaultValueColumns: []

18
17:52:09:09  documentSelectColumns: []

19
17:52:09:09  fieldTypes: {"test":"IMAGE","logoImage":"URL","symbols":"RICH_TEXT","abbreviated":"RICH_TEXT","descriptor":"SIMPLE_TEXT","solution":"SIMPLE_TEXT"}

20
17:52:09:09  extensions: []

21
17:52:09:09  runtimeOptions: {}

22
17:52:09:09  tableSchema: {"lastUpdatedGitHubUser":{"email":"bbm","username":"bbm@gmail"},"functionConfigPath":"_rowy_/settings/functions/keystrokes","columns":{"solution":{"fieldName":"solution","name":"solution","index":0,"type":"SIMPLE_TEXT","config":{},"key":"solution"},"descriptor":{"fieldName":"descriptor","name":"descriptor","index":1,"type":"SIMPLE_TEXT","config":{},"key":"descriptor"},"abbreviated":{"fieldName":"abbreviated","name":"abbreviated","index":2,"type":"RICH_TEXT","config":{},"key":"abbreviated"},"symbols":{"fieldName":"symbols","name":"symbols","index":3,"type":"RICH_TEXT","config":{},"key":"symbols"},"logoImage":{"fieldName":"logoImage","index":4,"type":"URL","config":{},"key":"logoImage","name":"image"},"test":{"fieldName":"test","type":"DERIVATIVE","key":"test","config":{"derivativeFn":"// Import any NPM package needed
// import _ from "lodash";

const derivative: Derivative = async ({ row, ref, db, storage, auth, logging }) => {
  logging.log("derivative started");

};

//write your code below here only

const url = row.image;
const image = await rowy.storage.upload.url(url);
ref.update({"image" : logo.image.downloadURL});

return [image];

//write your code above here only

export default derivative;","listenerFields":["logoImage"],"renderFieldType":"IMAGE"},"index":5,"name":"imagefile"}}}

23
17:52:10:10  Generating config file...

24
17:52:10:10  Retrieving config file...

25
17:52:10:10  Validating config file...

26
17:52:10:10  Executing: cd build/functionBuilder/builds/1694393512994/src;tsc functionConfig.ts

27
17:52:24:24  stdout: {"error":null,"stdout":"","stderr":""}

28
17:52:24:24  

29
17:52:24:24  Config file: "import fetch from "node-fetch";
import rowy from "./rowy";

export const fieldTypes = {
  "test": "IMAGE",
  "logoImage": "URL",
  "symbols": "RICH_TEXT",
  "abbreviated": "RICH_TEXT",
  "descriptor": "SIMPLE_TEXT",
  "solution": "SIMPLE_TEXT"
}
export const triggerPath = "keystrokes/{docId}"
export const functionName = "keystrokes"
export const derivativesConfig = [{
  fieldName: 'test',
  requiredPackages: [],
  // evaluate:require("./derivatives/test"),

  listenerFields: ["logoImage"]
}, ]
export const defaultValueConfig = []
export const documentSelectConfig = []
export const extensionsConfig = []
export const runtimeOptions = {
  "serviceAccount": "rowy-functions@keystrokes-ver1a.iam.gserviceaccount.com"
}
export const tableSchema = {
  "lastUpdatedGitHubUser": {
    "email": "bbm@gmail.com",
    "username": "bbma@gmail.com"
  },
  "functionConfigPath": "_rowy_/settings/functions/keystrokes",
  "columns": {
    "solution": {
      "fieldName": "solution",
      "name": "solution",
      "index": 0,
      "type": "SIMPLE_TEXT",
      "config": {},
      "key": "solution"
    },
    "descriptor": {
      "fieldName": "descriptor",
      "name": "descriptor",
      "index": 1,
      "type": "SIMPLE_TEXT",
      "config": {},
      "key": "descriptor"
    },
    "abbreviated": {
      "fieldName": "abbreviated",
      "name": "abbreviated",
      "index": 2,
      "type": "RICH_TEXT",
      "config": {},
      "key": "abbreviated"
    },
    "symbols": {
      "fieldName": "symbols",
      "name": "symbols",
      "index": 3,
      "type": "RICH_TEXT",
      "config": {},
      "key": "symbols"
    },
    "logoImage": {
      "fieldName": "logoImage",
      "index": 4,
      "type": "URL",
      "config": {},
      "key": "logoImage",
      "name": "image"
    },
    "test": {
      "fieldName": "test",
      "type": "DERIVATIVE",
      "key": "test",
      "config": {
        "derivativeFn": "// Import any NPM package needed
// import _ from "lodash";

const derivative: Derivative = async ({ row, ref, db, storage, auth, logging }) => {
  logging.log("derivative started");

};

//write your code below here only

const url = row.image;
const image = await rowy.storage.upload.url(url);
ref.update({"image" : logo.image.downloadURL});

return [image];

//write your code above here only

export default derivative;",
        "listenerFields": ["logoImage"],
        "renderFieldType": "IMAGE"
      },
      "index": 5,
      "name": "imagefile"
    }
  }
}
export const region = "us-west1"
export const searchHost = null
export const searchIndices = []"

30
17:52:24:24  Installing dependencies: []

31
17:52:24:24  Installing extensions: []

32
17:52:24:24  Installing dependencies...

33
17:52:25:25  [1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.

34
17:52:25:25  Deploying keystrokes to keystrokes-ver1a

35
17:52:25:25  Executing: cd build/functionBuilder/builds/1694393512994; yarn deploy --project keystrokes-ver1a --only functions

36
17:52:36:36  stdout: {"error":{"code":1,"killed":false,"signal":null,"cmd":"cd build/functionBuilder/builds/1694393512994; yarn deploy --project keystrokes-ver1a --only functions"},"stdout":"$ echo 'n
' | firebase deploy --interactive --project keystrokes-ver1a --only functions

=== Deploying to 'keystrokes-ver1a'...

i  deploying functions
i  functions: preparing codebase default for deployment
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
i  artifactregistry: ensuring required API artifactregistry.googleapis.com is enabled...
✔  functions: required API cloudfunctions.googleapis.com is enabled
✔  functions: required API cloudbuild.googleapis.com is enabled
✔  artifactregistry: required API artifactregistry.googleapis.com is enabled

Error: Failed to load function definition from source: Failed to generate manifest from function source: SyntaxError: await is only valid in async functions and the top level bodies of modules
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
","stderr":"error Command failed with exit code 1.
"}

37
17:52:36:36  $ echo 'n
' | firebase deploy --interactive --project keystrokes-ver1a --only functions

=== Deploying to 'keystrokes-ver1a'...

i  deploying functions
i  functions: preparing codebase default for deployment
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
i  artifactregistry: ensuring required API artifactregistry.googleapis.com is enabled...
✔  functions: required API cloudfunctions.googleapis.com is enabled
✔  functions: required API cloudbuild.googleapis.com is enabled
✔  artifactregistry: required API artifactregistry.googleapis.com is enabled

Error: Failed to load function definition from source: Failed to generate manifest from function source: SyntaxError: await is only valid in async functions and the top level bodies of modules
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

38
17:52:36:36  

command: cd build/functionBuilder/builds/1694393512994; yarn deploy --project keystrokes-ver1a --only functions

stderr: error Command failed with exit code 1.

errorStackTrace: Error: Command failed: cd build/functionBuilder/builds/1694393512994; yarn deploy --project keystrokes-ver1a --only functions
error Command failed with exit code 1.

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)

Ask AI to edit or generate...
Hyper

**Comment by Chadha93** [comment]
Hi @elizabethhannan, lets help you on this, thank you for sharing logs. Could you please record a video using jam.dev? And could you please share your Firebase project ID and Email ID.

I invite you to join our [Discord](https://www.rowy.io/discord) and post this on the community support channel. 

--
Cheers,
Gaurav Chadha | Technical Support Engineer | [Rowy](https://rowy.app/) + [BuildShip](https://buildship.com/)
backend for everyone

**Reply by Chadha93** [reply, marked as answer]
Feel free to re-open the discussion, if the issue still exists. We've recently launched Rowy [V3.0.0](https://github.com/rowyio/rowy/releases/tag/v3.0.0), which fixes :bug: bugs contains enhancements and new features.
